src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -897,4 +902,24 @@ public void setSpectatorTarget(@Nullable org.spongepowered.api.entity.Entity ent
         return event;
     }
 
+    @Override
+    public WorldBorder getWorldBorder() {
+        if (this.worldBorder == null) {
+            this.worldBorder = (WorldBorder) this.world.getWorldBorder();
+        }
+        return this.worldBorder;
+    }
+
+    @Override
+    public void setWorldBorder(WorldBorder border, Cause cause) {
+        WorldBorder originalBorder = this.worldBorder;
+        if (!SpongeImpl.postEvent(SpongeEventFactory.createChangeWorldBorderEventTargetPlayer(cause, originalBorder, border, this))) {
+            this.worldBorder = originalBorder;
```

Comment:
This should be the other way around. `postEvent` returns whether the event was cancelled.

Also, pretty sure you don't need to send `SPacketWorldBorder` if it wasn't changed